st2client/st2client/commands/action.py

```diff
@@ -503,6 +503,26 @@ def transform_object(value):
             return result
 
         def transform_array(value):
+            # Sometimes an array parameter only has a single element:
+            #
+            #     i.e. "st2 run foopack.fooaction arrayparam=51"
+            #
+            # Normally, json.loads can handle this, but not if the single element is
+            # an int.
+            #
+            # We want to try to handle this first, so it doesn't get accidentally
+            # sent to the API as an int, instead of an array of single-element int.
+            try:
+                # Force this to be a list containing the single int, then
+                # cast the whole thing to string so json.loads can handle it
+                value = str([int(value)])
```

Comment:
Another way we could improve handling of the array parameters in the future -

Users can also specify type for each array parameter inside an action metadata (`items: type: int` for example) and this information is available here so we could also leverage that to cast each array item to the correct type.

If nothing else we should probably just open a feature request issue for that change :)